### PR TITLE
button: support `icon-right` prop like `link`

### PR DESCRIPTION
**Pull request by CarterLi**
Please make sure these boxes are checked before submitting your PR, thank you!

* [ ] Make sure you follow Element's contributing guide ([中文](https://github.com/ElemeFE/element/blob/master/.github/CONTRIBUTING.zh-CN.md) | [English](https://github.com/ElemeFE/element/blob/master/.github/CONTRIBUTING.en-US.md) | [Español](https://github.com/ElemeFE/element/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/ElemeFE/element/blob/master/.github/CONTRIBUTING.fr-FR.md)).
* [x] Make sure you are merging your commits to `dev` branch.
* [x] Add some descriptions and refer relative issues for you PR.

### CODE REVIEW

File: packages/button/src/button-group.jsx [hunk 1]
```diff
@@ -0,0 +1,7 @@
+export default {
+  functional: true,
+  name: 'ElButtonGroup',
+  render(h, { data, children }) {
+    return <div {...data} class="el-button-group">{children}</div>;
```
Comment (YuhuiM) [inline packages/button/src/button-group.jsx:5]: 这用this.$slots.default比较合理一点

File: packages/button/src/button.vue [hunk 2]
```diff
@@ -15,9 +15,14 @@
...
-    <i class="el-icon-loading" v-if="loading"></i>
-    <i :class="icon" v-if="icon && !loading"></i>
-    <span v-if="$slots.default"><slot></slot></span>
+    <i
+      v-if="icon || loading"
+      :class="[
+        loading ? 'el-icon-loading' : icon,
+        $slots.default ? 'el-icon--left' : null
+      ]"></i>
+    <slot></slot>
```
Comment (YuhuiM) [inline packages/button/src/button.vue:24]: 缩进

File: packages/button/src/button-group.jsx (hunk 1 above)
Comment (CarterLi) [reply packages/button/src/button-group.jsx:5]: 这里用了函数式组件。函数式组件只有children没有this.$slots.default

File: packages/button/src/button.vue (hunk 2 above)
Comment (CarterLi) [reply packages/button/src/button.vue:24]: 这里缩进没问题，slot是对着上面的i的开始标签

**Pull request merged**